ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -61,15 +102,122 @@ export class ImportedDocuments extends React.Component {
         });
       } else {
         this.setState({
-          data: response.data,
           isLoading: true,
+          data: response.data,
         });
       }
+      this.calculateMode();
+      this.calculateStatus();
+      this.calculateStatistics();
     } else {
       this.intervalId && clearInterval(this.intervalId);
     }
   };
 
+  calculateMode = () => {
+    const { data } = this.state;
+    let { mode } = this.state;
+    switch (data.mode) {
+      case 'IMPORT':
+        mode = (
+          <Label color="blue" basic>
+            Import
+          </Label>
+        );
+        break;
+      case 'DELETE':
+        mode = (
+          <Label color="red" basic>
+            Delete
+          </Label>
+        );
+        break;
+      case 'PREVIEW_IMPORT':
+        mode = (
+          <Label color="teal" basic>
+            Preview (import)
+          </Label>
+        );
+        break;
+      case 'PREVIEW_DELETE':
+        mode = (
+          <Label color="teal" basic>
+            Preview (delete)
+          </Label>
+        );
+        break;
+      default:
+        mode = null;
+        break;
+    }
+    this.setState({
+      mode: mode,
+    });
+  };
+
+  statusLabel = () => {
+    const { data, isLoading } = this.state;
+    if (!data) return <Label color="grey">Fetching status</Label>;
+    if (isLoading)
+      return (
+        <Label color="blue">
+          <Icon loading name="circle notch" /> Importing literature
+        </Label>
+      );
+    switch (data.status) {
+      case 'SUCCEEDED':
+        return <Label color="green">Import successfull</Label>;
+      case 'CANCELLED':
+        return <Label color="yellow">Import cancelled</Label>;
+      default:
+        return <Label color="red">Import failed</Label>;
+    }
+  };
+
+  calculateStatus = () => {
+    this.setState({
+      status: this.statusLabel(),
+    });
+  };
+
+  calculateStatistics = () => {
+    const { statistics, data, selectedResult } = this.state;
+
+    const dataClone = Array.from(data.records);
```

Comment:
`Array.from()` creates a shallow copy i.e mutation to nested objects are still possible.

```javascript
let a = [{x:1,y:2,z:3}];
let b = Array.from(a);
b[0].x = 0;
console.log(JSON.stringify(a)); // [{"x":0,"y":2,"z":3}]
console.log(JSON.stringify(b)); // [{"x":0,"y":2,"z":3}]
```
You can use lodash https://lodash.com/docs/4.17.15#cloneDeep function instead!